packages/terra-functional-testing/src/services/wdio-terra-service.js

```diff
@@ -144,7 +148,30 @@ class TerraService {
    */
   async onComplete(_, config) {
     try {
-      if (this.serviceOptions.useRemoteReferenceScreenshots && !this.serviceOptions.buildBranch.match(BUILD_BRANCH.pullRequest) && this.serviceOptions.buildType === BUILD_TYPE.branchEventCause) {
+      if (this.serviceOptions.useRemoteReferenceScreenshots && process.env.SCREENSHOT_MISMATCH_CHECK && this.serviceOptions.buildBranch.match(BUILD_BRANCH.pullRequest)) {
```

Comment:
Originally I assumed that since we were manually making sure that mismatch failures counted as a success, any passed results would also be a success. Following that I assumed that I would have to search through all the tests' messages to see if the "Screenshot has changed and needs to be reviewed." string was present to detect, which seemed a tad too inefficient.

According to that documentation `onComplete` should run with 4 arguments with the 4th argument being a "results" object. When I console.log that 4th argument I get `undefined` as the answer. 

I also tried using the 1st argument, the exit code, but it reports success even when I purposely put in a failure screenshot but force `toMatchReference` calls to follow the remote reference screenshot logic in it.

Reply:
I would think the results should be returned in the the parameter. I think the current implementation would work but just feels a bit hacky. I guess we can keep the current implementation if we don't know why the returned result is not what we expected.

Reply:
Well the other concern is, aren't we marking the test a pass anyway? So even if we got the results the only way to know if a test failed due to mismatch is to read through all the messages for each test and find at least one that has either our special added message or the mismatch message?